testing/steps/test_zip_data.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+
+from neuraxle.base import ExecutionContext
+from neuraxle.data_container import DataContainer
+from neuraxle.pipeline import Pipeline
+from neuraxle.steps.data import ZipData
+
+TIMESTEPS = 10
+FEATURES = 5
+VALIDATION_SIZE = 0.1
+BATCH_SIZE = 32
+N_EPOCHS = 10
+SHAPE_3D = (BATCH_SIZE, TIMESTEPS, FEATURES)
+SHAPE_2D = (BATCH_SIZE, TIMESTEPS)
+SHAPE_1D = BATCH_SIZE
+
+
+def test_zip_data_should_merge_3d_with_3d():
+    # Given
+    data_inputs_3d, expected_outputs_3d = _create_data_source(SHAPE_3D)
+    data_inputs_3d_second, expected_outputs_3d_second = _create_data_source(SHAPE_3D)
+    data_container_3d_second = DataContainer(data_inputs=data_inputs_3d_second, expected_outputs=expected_outputs_3d_second)
+    data_container = DataContainer(data_inputs=data_inputs_3d, expected_outputs=expected_outputs_3d) \
+        .add_sub_data_container('2d', data_container_3d_second)
+
+    # When
+    p = Pipeline([
+        ZipData(data_sources=['2d'])
+    ])
+
+    data_container = p.handle_transform(data_container, ExecutionContext())
+
+    # Then
+    assert data_container.data_inputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] * 2)
+    assert data_container.expected_outputs.shape == (SHAPE_3D[0], SHAPE_3D[1], SHAPE_3D[2] * 2)
+    assert np.array_equal(data_container.data_inputs[..., -SHAPE_3D[2]:], data_container_3d_second.data_inputs)
+    assert np.array_equal(data_container.expected_outputs[..., -SHAPE_3D[2]:], data_container_3d_second.expected_outputs)
+
+
+def test_zip_data_should_merge_2d_with_3d():
+    # Given
+    data_inputs_3d, expected_outputs_3d = _create_data_source(SHAPE_3D)
+    data_inputs_2d, expected_outputs_2d = _create_data_source(SHAPE_2D)
+    data_container_2d = DataContainer(data_inputs=data_inputs_2d, expected_outputs=expected_outputs_2d)
+    data_container_3d = DataContainer(data_inputs=data_inputs_3d, expected_outputs=expected_outputs_3d) \
+        .add_sub_data_container('2d', data_container_2d)
```

Comment:
I dont like that it is one main and one sub container. Would like to see a containerwrapper that contains 2 subs instead.

Reply:
E.g.: Do like the expand dims

Reply:
E.g.: create an empty one and add 2 subs.

Reply:
https://github.com/Neuraxio/Neuraxle/blob/05b12db/neuraxle/steps/flow.py#L464

It uses the sub ones as a data input.